libraries/Microsoft.Bot.Builder.Core/IBotContext.cs

```diff
@@ -17,66 +17,63 @@ public interface IBotContext
         /// Incoming request
         /// </summary>
-        IActivity Request { get; }
+        Activity Request { get; }
 
         /// <summary>
         /// Respones
         /// </summary>
-        IList<IActivity> Responses { get; set; }
+        IList<Activity> Responses { get; set; }
 
         /// <summary>
         /// Conversation reference
         /// </summary>
         ConversationReference ConversationReference { get; }
 
-        /// <summary>
-        /// Bot state 
-        /// </summary>
-        BotState State { get; }
-    
-        Intent TopIntent { get; set; }
-
-        /// <summary>
-        /// check to see if topIntent matches
-        /// </summary>
-        /// <param name="intentName"></param>
-        /// <returns></returns>
-        bool IfIntent(string intentName);
-
-        /// <summary>
-        /// Check to see if intent matches regex
-        /// </summary>
-        /// <param name="expression"></param>
-        /// <returns></returns>
-        bool IfIntent(Regex expression);
-
         /// <summary>
         /// Queues a new "message" responses array.
         /// </summary>
         /// <param name="text">Text of a message to send to the user.</param>
         /// <param name="speak">(Optional) SSML that should be spoken to the user on channels that support speech.</param>
         /// <returns></returns>
-        BotContext Reply(string text, string speak = null);
+        IBotContext Reply(string text, string speak = null);
 
         /// <summary>
         /// Queues a new "message" responses array.
         /// </summary>
         /// <param name="activity">Activity object to send to the user.</param>
         /// <returns></returns>
-        BotContext Reply(IActivity activity);
+        IBotContext Reply(IActivity activity);
 
-    }   
+        /// <summary>
+        /// Set named service
+        /// </summary>
+        /// <param name="serviceId"></param>
+        /// <param name="service"></param>
+        void Set(string serviceId, object service);
+
+        /// <summary>
+        /// Get named service
+        /// </summary>
+        /// <param name="serviceId"></param>
+        /// <returns>service</returns>
+        object Get(string serviceId);
+    }
 
     public static partial class BotContextExtension
     {
-        //public static async Task Send(this BotContext context)
-        //{            
-        //    await context.SendActivity(context, new List<IActivity>()).ConfigureAwait(false);
-        //}
+        public static void Set<ServiceT>(this IBotContext context, ServiceT service, string serviceId = null)
+        {
+            if (serviceId == null)
+                serviceId = $"{typeof(ServiceT).Namespace}.{typeof(ServiceT).Name}";
+            context.Set(serviceId, service);
+        }
 
-        public static BotContext ToBotContext(this IBotContext context)
+        public static ServiceT Get<ServiceT>(this IBotContext context, string serviceId = null)
         {
```

Comment:
Not sure why this is an Extension method, rather than just on the actual class. Seems like these Generic methods should just be on the main class, and this class goes away.